### PR TITLE
Recyclerview

### PR DESCRIPTION
**Pull request by ragore09**

### CODE REVIEW

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import wasp.com.recyclerviewexample.adapter.UserAdapter;
+import wasp.com.recyclerviewexample.model.User;
+
+public class MainActivity extends AppCompatActivity {
+
+    @BindView(R.id.rvUsers)
+    RecyclerView rvUsers;
```
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/MainActivity.java]: @ragore09 try to [follow this code style](https://source.android.com/source/code-style.html) when possible. So you should name this as `mRVUsers` or `mUsersRecyclerView`.

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import wasp.com.recyclerviewexample.R;
+import wasp.com.recyclerviewexample.model.User;
+
+public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {
+
+    private List<User> users;
```
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: @ragore09 you could name this as `mUsers`

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        @BindView(R.id.user_username)
+        public TextView userNameTextView;
+
+        @BindView(R.id.user_name)
+        public TextView userTextView;
+
+        @BindView(R.id.message_button)
+        public Button messageButton;
+
+        private Context context;
```
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: @ragore09 why do you need this context reference?

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import wasp.com.recyclerviewexample.R;
+import wasp.com.recyclerviewexample.model.User;
+
+public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {
+
+    private List<User> users;
+    private Context context;
```
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: @ragore09 why do you need this context reference? you could remove it.

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        ViewHolder viewHolder = new ViewHolder(context, userView);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder viewHolder, int position) {
+        User user = users.get(position);
+
+        TextView textView = viewHolder.userNameTextView;
```
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: @ragore09 you could create a `bind(User user) {}` method in your `ViewHolder` class and move all this logic there to just call `viewHolder.bind(user);`

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/MainActivity.java (hunk 1 above)
Comment (ragore09) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/MainActivity.java]: ok! i'll fix that

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java (hunk 2 above)
Comment (ragore09) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: the letter `m`, what stands for?

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java (hunk 3 above)
Comment (ragore09) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: i'm using this context for binding the views with butter knife inside the ViewHolder constructor `ButterKnife.bind(this, itemView);`, is this correct? or how should it be done?

Comment (ragore09) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: I found the workaround!

File: RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java (hunk 2 above)
Comment (epool) [inline RecyclerViewExample/app/src/main/java/wasp/com/recyclerviewexample/adapter/UserAdapter.java]: it's for private class fields, is part of the android code styles, but it's not mandatory, it was just a suggestion.

**Pull request merged**